crates/sui-json-rpc/src/governance_api.rs

```diff
@@ -117,36 +119,28 @@ impl GovernanceReadApi {
         });
 
         if let Some(active_rate) = active_rate {
-            return Ok(active_rate);
+            Ok(active_rate)
         } else {
             // try find from inactive pool
-            let inactive_validators = system_state.inactive_pools_id;
-            let inactive_validators = self
-                .state
-                .read_table_value::<ObjectID, ValidatorV1>(inactive_validators, pool_id)
-                .await;
-            if let Some(inactive_validators) = inactive_validators {
-                return Ok(inactive_validators.staking_pool.exchange_rates.id);
-            }
+            let validator = get_validator_from_table(
+                system_state.system_state_version,
+                self.state.db().as_ref(),
+                system_state.inactive_pools_id,
+                &ID { bytes: *pool_id },
```

Comment:
I've seen this pattern a few times now (`ID` from `ObjectID`) -- thoughts on an `impl From<ObjectID>`?